101-service-accounts/proposal.md

```diff
@@ -0,0 +1,75 @@
+# Summary
+
+Teams should be able to define a list of service accounts during `fly set-team`
+that can be used in automation.
+
+
+# Motivation
+
+While Concourse is very good at automating things, it's not so good at
+automating itself. We've made progress towards it via things like the
+[set_pipeline step][set-pipeline-step] and the future [projects RFC][projects].
+However, there are still advanced use cases (like pausing pipelines, jobs,
+onboarding teams, pinning resources, etc) which are not solved by an RFC.
+
+Many Concourse operators end up creating local users to be used in scripts to
+setup Concourse. However, this presents a security risk since these accounts
+usually have god powers as the owner of the main team. It's also difficult to
+rotate local users' passwords since it requires a restart of the web node.
+
+# Proposal
+
+Multiple service accounts can be created at the team level and be explicitly
```

Comment:
> Allowing service accounts to bind to multiple teams would make them more similar to how users operate

If you're talking about having to explicitly configure per team, then users also have to be explicitly bound to each team. If you're talking about being able to seamlessly auth to different teams, then I think they would be able to have a similar experience by using the same token (i.e. using the same secret path) for multiple teams. Although that could be a security no-no.

> if you need to configure things per-team, you can have a single admin service account, rather than needing to manage one per team.

A service account configured under the `main` team should have this power.

Reply:
I wonder if there's scenarios where there's non-`main` teams managing other teams. My current experience with multi-tenancy is that there's a cluster operator team with god mode managing the `main` team, and then sub teams are only 1 level deep.
```
main
├── team-1
└── team-2
```
I wonder if there's cases where there's sub-teams in charge of sub-teams
```
main
├── operator-1
│   ├── operator-2
│   │   └── ...
│   ├── team-2
│   └── team-3
└── team-1
```